common/lib/container-definitions/src/runtime.ts

```diff
@@ -97,7 +97,7 @@ export interface IRuntime extends IDisposable {
      * using IDs generated locally. After attach, these IDs cannot be used, so this table maps the old local IDs to the
      * new storage IDs so requests can be redirected.
      */
-    createSummary(blobRedirectTable?: Map<string, string>): ISummaryTree;
```

Comment:
I want to make sure it works correctly, and comments to reflect expectations.
This API should return state of container at the time when this call was initiated, even though promise may take a minute to get completed.
Given that it's super easy to create bugs here with async workflow (because user edits in between start of this call and when promise is completed can get into snapshot and thus make snapshot incorrect), so I'd expect to see extensive tests validating various types of changes (DDS, but also structural, i.e., creation of DDS / data stores) in between those two points never get into snapshot.